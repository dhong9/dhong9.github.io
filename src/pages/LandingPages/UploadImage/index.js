/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

// react-router-dom components
import { useState } from "react";

// Material Kit 2 React components
import MKBox from "components/MKBox";
import MKButton from "components/MKButton";
import MKTypography from "components/MKTypography";

// Material Kit 2 React examples
import MKAvatar from "components/MKAvatar";

// Image
import defaultProfileImage from "assets/images/default_profile.jpg";

function UploadImage() {
  // User could not have gotten to this component without a regular login

  const [selectedFile, setSelectedFile] = useState(defaultProfileImage);

  const handleImageUpload = (e) => {
    const [file] = e.target.files;
    const reader = new FileReader();
    reader.readAsDataURL(file);

    reader.onloadend = (e) => {
      console.log("Event: ");
      console.log(e);
      setSelectedFile(reader.result);
    };
    console.log(selectedFile);
  };

  return (
    <>
      <MKBox
        variant="gradient"
        bgColor="info"
        borderRadius="lg"
        coloredShadow="info"
        mx={2}
        mt={-3}
        p={2}
        mb={1}
        textAlign="center"
      >
        <MKTypography variant="h4" fontWeight="medium" color="white" mt={1}>
          Upload Image
        </MKTypography>
      </MKBox>
      <MKBox pt={4} pb={3} px={3}>
        <MKBox component="form" role="form">
          <MKBox mb={2} justifyContent="center" display="flex">
            <MKAvatar src={defaultProfileImage} alt="Profile picture" size="xxl" />
          </MKBox>
          <MKBox mb={2}>
            <MKButton variant="contained" component="label">
              Select Image
              <input
                id="contained-button-file"
                type="file"
                accept="image/*"
                onChange={handleImageUpload}
                multiple
                hidden
              />
            </MKButton>
          </MKBox>
        </MKBox>
      </MKBox>
    </>
  );
}

export default UploadImage;
