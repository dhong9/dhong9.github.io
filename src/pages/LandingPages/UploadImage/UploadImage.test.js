/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

// React testing libraries
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

// Material Kit 2 React themes
import { ThemeProvider } from "@mui/material/styles";
import theme from "assets/theme";

// Authentication
import UploadImage from "pages/LandingPages/UploadImage";

describe("UploadImage", () => {
  it("renders", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <UploadImage user={null} updateUser={jest.fn} />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
