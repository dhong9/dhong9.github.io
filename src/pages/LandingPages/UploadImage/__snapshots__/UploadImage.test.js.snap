// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImage renders 1`] = `
<div>
  <div
    class="MuiBox-root css-15m7mkq"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-o30d1n-MuiTypography-root"
    >
      Upload Image
    </h4>
  </div>
  <div
    class="MuiBox-root css-1iu09ww"
  >
    <form
      class="MuiBox-root css-1ircn5c"
      role="form"
    >
      <div
        class="MuiBox-root css-16pyyvj"
      >
        <span
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1kav12i-MuiButtonBase-root-MuiButton-root"
          role="button"
          tabindex="0"
        >
          Select Image
          <div
            accept="image/*"
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1c3j9he-MuiFormControl-root-MuiTextField-root"
            hidden=""
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-uk4jy3-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Image
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-12n1zae-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-5mmmz-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                type="file"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-135f4hd-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Image
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
      </div>
    </form>
  </div>
</div>
`;
